lib/core/presentation/pages/scan_qr_code/views/scan_qr_code_view.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:app/core/application/event/update_event_checkin_bloc/update_event_checkin_bloc.dart';
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/scan_qr_code_page.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/widgets/scanner_actions.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/widgets/scanner_error_widget.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/widgets/scanner_overlay.dart';
+import 'package:app/core/utils/snackbar_utils.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/router/app_router.gr.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+
+class ScanQRCodeView extends StatefulWidget {
```

Comment:
This page seems couple with reward feature, if you cannot do it generic, then lets rename with specific use case , and move to event setting rewards page